simdjson-flags.cmake

```diff
@@ -39,7 +39,7 @@ add_library(simdjson-internal-flags INTERFACE)
 target_link_libraries(simdjson-internal-flags INTERFACE simdjson-flags)
 
 if(MSVC)
-  target_compile_options(simdjson-internal-flags INTERFACE /WX /W3 /sdl)
+  target_compile_options(simdjson-internal-flags INTERFACE /WX /W3 /sdl /utf-8)
```

Comment:
It looks like this must be on by default in MSVC? I'm worried about adding flags that *prevent* errors, simply because then people who compile our .cpp/.h straight up may get those errors.

Can we prefix the affected file with a comment or preamble (like the headers emacs can detect, or the UTF-8 byte order) that tells MSVC it's UTF-8? In the end, if no other solution can be found, I'd rather move the wonderful UTF-8 comments into a doc/utf8-design.md than add the flag.

Reply:
A better solution, in my opinion, is to code using strict ASCII. For what we are trying to do in this library, it should be fine.

Reply:
Yeah, I'm fine with putting the UTF-8 design comments into a .md. I'd definitely rather do that than add a flag.

Reply:
> Yeah, I'm fine with putting the UTF-8 design comments into a .md.

Done in another PR.

Reply:
Confirm fixed in #899. I will remove the first commit.